marklogic-client-api/src/main/java/com/marklogic/client/impl/RowManagerImpl.java

```diff
@@ -285,6 +285,30 @@ public RawSPARQLSelectPlanImpl newRawSPARQLSelectPlan(TextWriteHandle handle) {
     return handle.get();
   }
 
+  @Override
+  public <T extends JSONReadHandle> T columnInfo(Plan plan, T resultsHandle) {
```

Comment:
Should the type of the first parameter be Plan or PreparePlan?

PlanBuilder.map() returns ExportablePlan. Does column info return useful information for ExportablePlan?  If so, Plan is the right type.

Otherwise, consistent with the unit test, PreparePlan is the right type.

Reply:
Hi Erik, I'm a little confused here. My unit test used PlanBuilder.ExportablePlan and it works, so I think columnInfo() works with ExportablePlan.  And what is PlanBuilder.map()? Thanks.

Reply:
Hi, Ling:  Sorry, I meant PlanBuilder.PreparePlan.map​() -- the client method equivalent to protptype.map() on the enode.

In the unit test, is the actual type of the object ExportablePlan?  Or is the actual type of the object a super interface of ExportablePlan assigned to a variable of type ExportablePlan?  I ask because the actual type of the object and not the type of the assigned variable is what matters for validation.

I ask because my guess is that column info can't be returned on the plan generated by map(), given that map could return anything.  If that's correct, then the parameter shouldn't accept the plan objects constructed by PlanBuilder.PreparePlan.map​()